### ISSUE TITLE
Respect CategoricalArray order in `group` kwarg

### ISSUE THREAD

**Issue by kescobo** [JuliaPlots/Plots.jl#1712]
```julia
l = CategoricalArray(repeat(["a", "c", "b"], 3))
levels!(l, ["a", "c", "b"])
c = [:red :blue :green]

scatter(rand(9), rand(9),
    group=l, # I want "c" to go second
    color=c) # I want "c" to be blue
```

![levels1](https://user-images.githubusercontent.com/3502975/44809412-a23db100-ab9c-11e8-8892-f06a82f622e1.png)

Instead, "c" should be second, and be blue:

```
scatter(rand(9), rand(9), primary=false, color=repeat(vec(c), 3))
for i in eachindex(levels(l))
    scatter!([],[], label=levels(l)[i], color=c[i])
end
```

![levels2](https://user-images.githubusercontent.com/3502975/44809575-1415fa80-ab9d-11e8-9009-1851fc784d13.png)

**Comment by mkborregaard** [comment]
Agreed, that should be the behaviour.

**Comment by kescobo** [comment]
Looking back at this - sorry to be so terse! There was a discussion in slack that led to it. Don't want to seem like I'm just coming in here and being demanding 😄

**Comment by mkborregaard** [comment]
Nonono, you're being a good open source citizen by reporting it and I do agree :-)

**Comment by pgbarletta** [comment]
I was about to open an issue about this same topic. In my case, I'm having trouble with `groupedbar()` so I can't `plot()` and then `plot!()`. Unless there is some other way of bypassing this. Is there?

**Comment by kescobo** [comment]
@pgbarletta Can you explain a bit more about what you're trying to do with grouped bar? I'm not sure, but since that's part of `StatPlots` (and has some nuances since it's a user recipe rather than a series recipe), it might be worth opening a separate issue over there.

**Comment by pgbarletta** [comment]
I actually made an example before finding this issue. Hope it's clear enough:

https://nbviewer.jupyter.org/github/pgbarletta/lbp/blob/master/src/Untitled.ipynb

**C1-CG** and **O1, O2** are atom names, so my molecule is getting reordered :)

Let me know if I should open an issue on `StatPlots`. Thanks for the help.

**Comment by kescobo** [comment]
Thanks for the example @pgbarletta. I think your case is a bit different - this issue is about [`CategoricalArrays`](https://github.com/JuliaData/CategoricalArrays.jl), and specifically about the order of the groups, (in your case `V1` and `V2`) rather than the order on the xaxis. 

In your case, you can just use numbers for ordering and then set the labels:

```
groupedbar(repeat(1:18, 2), hcat(v1, v2),
    group = g_lig, xaxis=("Atoms", (1:18, x_lig)))
```

**Comment by pgbarletta** [comment]
You're right, it worked! I didn't know `xaxis` could take a `Tuple` and assign labels to the ticks.
Thanks a lot.

https://nbviewer.jupyter.org/github/pgbarletta/lbp/blob/master/src/Untitled.ipynb

**Comment by WMOkiishi** [comment]
It seems that this now *does* just work. However, there's one way in which it still doesn't quite do what I'd like it to.
```jl
l = CategoricalArray(repeat(["a", "c", "b"], 3))
levels!(l, ["a", "c", "d", "b"])  # Note the addition of an unused "d" value here
c = [:red :blue :orange :green]

scatter(rand(9), rand(9),
    group=l,  # Order should be a, c, (d,) b -- d is skipped, but the order is otherwise respected 
    color=c)  # Whether any "d" values are present or not, I'd like "b" to be green -- it's orange instead
```
I haven't been able to find an elegant workaround. The best I've come up with is using recipes with my own custom keywords for grouping and colors. As far as I can tell, since grouping is handled before any custom recipes I can add, there's no way for me to hook into that process.
